File: DesignPatternsExercises/java/BuilderDesignPattern/Solution/BuilderDesignPattern.java
public class BuilderDesignPattern {
    public static void main(String[] args) {
        ComputerEngineer computerEngineer = new ComputerEngineer();

        Computer computer = computerEngineer.buildComputer();
        PCBuilder pcBuilder = computerEngineer.getComputerBuilder();
        pcBuilder.addHardDisk(64).addRam(256).addMonitor(17).addMonitor(28).addHardDisk(512).addHardDisk(1024).addHardDisk(2048).addMouse("Optical");

        System.out.println("Computer is assembled and with the following parts.");
        System.out.println("List of Monitors:");
        for (Monitor monitor : computer.getMonitorList())
            System.out.println("    " + monitor.getSize() + " inches");

        System.out.println("List of RAMs:");
        for (Ram ram : computer.getRamList())
            System.out.println("    " + ram.getRamCapacity() + " MB");

        System.out.println("Total Ram Capacity: " + computer.getRamCapacity() + " MB");

        System.out.println("List of HDDs:");
        for (HardDisk hardDisk : computer.getHddList())
            System.out.println("    " + hardDisk.getHardDiskCapacity() + " GB");

        System.out.println("Total HDD Capacity: " + computer.getHddCapacity() + " GB");

        if (computer.getMouse() == null)
            System.out.println("Computer Mouse Type: NONE");
        else
            System.out.println("Computer Mouse Type: " + computer.getMouse().getMouseType());
    }
}

File: DesignPatternsExercises/ruby/builder_design_pattern/exercise/builder_design_pattern.rb
require_relative 'computer_engineer'

computer_engineer = ComputerEngineer.new

computer = computer_engineer.make_computer
computer_builder = computer_engineer.computer_builder
computer_builder.add_hard_disk(64).add_ram(256).add_monitor(17).add_monitor(28).add_hard_disk(512).add_hard_disk(1024).add_hard_disk(2048).add_mouse('Optical')

puts 'Computer is assembled and with the following parts.'
puts 'List of Monitors:'
computer.monitor_list.each { |monitor| puts "    #{monitor.size} inches" }

puts 'List of RAMs:'
computer.ram_list.each { |ram| puts "    #{ram.capacity} MB" }

puts "Total Ram Capacity: #{computer.ram_capacity} MB"

puts 'List of HDDs:'
computer.hdd_list.each { |hdd| puts "    #{hdd.capacity} GB" }

puts "Total HDD Capacity: #{computer.hdd_capacity} GB"

computer.mouse ? (puts "Computer Mouse Type: #{computer.mouse.type}") : (puts 'Computer Mouse Type: NONE')
